### DISCUSSION TITLE
Handling specific units

### DISCUSSION THREAD

**Discussion by tpluscode** [qudt/qudt-public-repo#979] [Q&A]
Hello. I am curating some datasets for a customer and we determined a set of unit we need to annotate their data. Here are those which I'm unsure how to handle and would kindly ask for guidance.

**Million units**

Would I derive this from `qudt:NUM` and store locally or submit a `qudt:MegaNUM`?

**CHF / kg** 
**CHF / 100 kg**
**Centime / kg**

`qudt:CHF-PER-KiloG` already exists. Would I also add the derived units to QUDT?

**CHF / Liter**

Would I add `qudt:CHF-PER-L`?

**Centime / Unit**
**Eurocent / Unit**

What about these "currency per unit" units? Would I add some `CHF-PER-Unit` and `EUR-PER-Unit` and the derived units for the 1/100 part of CHF/EUR?

**Unit / Person**
**kg / Person**

Also, I did not find any similar units in QUDT. How should I define such "per person" units?

**"kg fresh meat equivalent"**

This one I suppose does not belong to QUDT at all. Would I derive it as being 1-1 with `qudt:KiloGM`?

Sorry if that's a lot of questions and thanks in advance.

**Comment by steveraysteveray** [comment]
Since the number of prefixed units and other derivations is enormous, our policy is to add (or have you add via a PR!) such units as requested. So, we would welcome a PR where you add:

- MegaNUM
- CHF-PER-HectoKiloGM (note, not KiloG which has to do with the unit of gravitational acceleration) I realize now that the topic of prefixes that fall outside of the standard ones was discussed in several Issues and Pull Request (such as [this one](https://github.com/qudt/qudt-public-repo/pull/958)), but we never documented our solution in our Wiki. Noted.
- Centime and Eurocent are interesting. Certainly those names are more recognizable, compared to DeciCHF and DeciEUR. I'd be interested in hearing other opinions on those.
- I notice we have been a little inconsistent on the dimensionless NUM vs. COUNT, where we use NUM-PER... but also have ...-PER-KiloCount. I think we should stick with one consistent unit, either NUM or COUNT. Either way I think we should avoid the term Unit, since that already means something very different. So we could have NUM-PER-INDIV. (We use INDIV rather than Person since it is more general). Thoughts?
- Finally, you are correct that "kg fresh meat equivalent" falls to you to define a qudt:Quantity as such, with a unit of KiloGM. We do handle some "equivalent" concepts as quantity kinds (e.g. quantitykind:EquivalentConcentration) but they are more general than your example.

**Comment by steveraysteveray** [comment]
I'm circling back on this topic with some updates:
 - We now include most common counting units (unit:MILLION, unit:THOUSAND, etc) as instances of both qudt:Unit and qudt:Counting Unit. These counting units use the qudt:scalingOf unit:ONE triple.
 - As of [Release 3.1.0](https://github.com/qudt/qudt-public-repo/releases/tag/v3.1.0), all of our currency URIs are now named with the pattern "CCY_<currency-code>" such as unit:CCY_CHF-PER-KiloGM. Additional combinations will be reviewed upon request via PR.
 - The "per person" units exist, but have been generalized to "individual", as in unit:DEATHS-PER-KiloINDIV-YR for quantitykind:MortalityRate.
 - The currency per currency units are still under discussion. Definitely pros and cons here.

**Comment by jhodgesatmb** [comment]
Must have missed the discussion about adding "CCY_" to the beginning of all currencies. Can you update the rationale for this because at a glance it doesn't seem necessary and makes one wonder why other 'domain-specific' additions wouldn't then be suggested (which would also seem unnecessary). Also, the underscore is generally a designator for a qualifier, but what is following the "CCY" is not a qualifier but a unit name. This is confusing.

**Reply by steveraysteveray** [reply]
What follows the "CCY_" is no longer a unit name - that is what has been deprecated. It is, in fact, the qualifier of a currency.

One reason we did this was the potential for collisions between currency codes and other unit URIs. For example, unit:AWG versus unit:CCY_AWG.

Not sure what you mean by 'domain-specific' additions.

**Reply by jhodgesatmb** [reply]
You are saying that the "CHF" in "unit:CCY_CHF" is a qualifier and not a currency? In the past it was a currency.

If everything past the "CCY_" is a qualifier then my last comment (i.e., 'domain-specific') is moot. That said, it was a reference to the "CCY" being a reference to currencies as a domain-specific designation.

**Reply by fkleedorfer** [reply]
@jhodgesatmb are you suggesting we should have chosen `_CCY` as a suffix, such as `CHF_CCY` and `EUR_CCY`?
